Library/ViewModels/PledgeViewModel.swift

```diff
@@ -338,6 +361,70 @@ public class PledgeViewModel: PledgeViewModelType, PledgeViewModelInputs, Pledge
     let createBackingTransactionSuccess = project.takeWhen(createBackingEventSuccess)
 
     self.goToThanks = Signal.merge(applePayTransactionCompleted, createBackingTransactionSuccess)
+
+    let amountChanged = Signal.combineLatest(
+      project.map { $0.personalization.backing?.pledgeAmount },
+      self.pledgeAmountDataSignal.map { $0.0 }
+    )
+    .map(!=)
+
+    let shippingRuleChanged = Signal.merge(
+      self.viewDidLoadProperty.signal.mapConst(false),
+      Signal.combineLatest(
+        project.map { $0.personalization.backing?.locationId },
```

Comment:
Here you could use `backing.map { $0.locationId }` can't wait for key paths to be supported in `.map` 😅